src/engines/L3/element.js

```diff
@@ -298,23 +298,13 @@ const Element = {
     this.props[prop] = unpackTransition(value)
     const props = Object.entries(this.props.props)
 
-    if (props.length === 1) {
-      const [p, v] = props[0]
+    for (let i = 0; i < props.length; i++) {
+      const [p, v] = props[i]
       if (isTransition(value)) {
         return this.animate(p, v, value.transition)
       } else if (this.node[p] !== v) {
         this.node[p] = v
       }
-    } else {
-      for (let i = 0; i < props.length; i++) {
-        // todo: fix code duplication
-        const [p, v] = props[i]
-        if (isTransition(value)) {
-          return this.animate(p, v, value.transition)
-        } else if (this.node[p] !== v) {
-          this.node[p] = v
-        }
-      }
     }
```

Comment:
I prefer leaving this unchanged, at least in this PR.

While your change works and makes sense (in fact it used to be exactly like this), from some (preliminary) tests it seems like the current code is more optimal from a JIT perspective.

In most cases there will be only 1 prop to be updated. Creating a new scope with a for loop even when there's on 1 item to loop over is then suboptimal.

We should benchmark this better though, but for now let's keep the construct as it is.

Reply:
Reverted